### ISSUE TITLE
[Bug] Creating Galera cluster using existing databases

### ISSUE THREAD

**Issue by kenlasko** [mariadb-operator/mariadb-operator#1028]
**Documentation**
- [X] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
I'm testing the MariaDB operator to replace my Bitnami Galera cluster that is running on Talos OS using local node storage for the replicas.

I'm testing a DR workflow where my Galera cluster gets destroyed for whatever reason, but the database files themselves are still available. In this scenario, the namespace hosting the MariaDB operator has been deleted. 

In my tests, when I rebuild the cluster, I had hoped the cluster could just pick up the databases where they were left off, but I get numerous errors around connection issues to other nodes, which I suspect may be due to some secure authentication that is uniquely generated for each new cluster that the original databases are unaware of.

Testing my other scenario where I bootstrap from a backup works just fine. 

**Expected behaviour**
I would like to be able to build a Galera cluster using a set of existing databases on the nodes. I should be able to pick an existing database as the initial seed to build the other replicas. 

**Steps to reproduce the bug**
1. Install MariaDB Operator and build Galera cluster using backup. Ensure functionality
2. Delete the MariaDB Operator namespace
3. Re-install MariaDB Operator and attempt to rebuild Galera cluster without backup bootstrapping
4. Note connection errors

**Debug information**
Can retrieve if necessary. Cluster currently deleted until next testing round.

**Environment details**:
- Kubernetes version: 1.31.2
- Kubernetes distribution: Talos 1.8.2
- mariadb-operator version: 0.36.0
- Galera data-plane version (init and agent): Unsure
- Install method: helm

**Comment by kenlasko** [comment]
I've finally had time to spend on this and am wondering if I'm just doing something wrong. Here's my workflow:
1. Create Galera cluster using `bootstrapFrom` - This works fine
2. Simulate total Kubernetes disaster by deleting the MariaDB resource followed by deleting the entire operator and namespace
3. Attempt to rebuild the cluster
   a. Comment out `bootstrapFrom` section from MariaDB definition
   b. Rely on the following section to initiate recovery:
```
  galera:
    enabled: true
    primary:
      podIndex: 0
      automaticFailover: true
    sst: mariabackup
    replicaThreads: 1
    config: 
      reuseStorageVolume: true
    agent:
      kubernetesAuth:
        enabled: true
      gracefulShutdownTimeout: 5s
    recovery:
      enabled: true
      minClusterSize: 1
      # forceClusterBootstrapInPod: "mariadb-0"
      clusterMonitorInterval: 10s
      clusterHealthyTimeout: 45s
      clusterBootstrapTimeout: 20m0s
      podRecoveryTimeout: 5m0s
      podSyncTimeout: 5m0s
```

When I recreate the cluster, I see the 3 pods start up and immediately fail with these errors:
```
2024-11-08 17:51:49 0 [Note] WSREP: (171c7ea0-adaa, 'tcp://0.0.0.0:4567') connection to peer 00000000-0000 with addr tcp://10.244.5.80:4567 timed out, no messages seen in PT3S, socket stats: rtt: 0 rttvar: 250000 rto: 1000000 lost: 1 last_data_recv: 957568792 cwnd: 1 last_queued_since: 957868925418913 last_delivered_since: 957868925418913 send_queue_length: 0 send_queue_bytes: 0
2024-11-08 17:51:49 0 [Note] WSREP: Failed to establish connection: Operation aborted.
2024-11-08 17:51:50 0 [Note] WSREP: Failed to establish connection: Connection refused
2024-11-08 17:51:51 0 [Note] WSREP: Failed to establish connection: Connection refused
2024-11-08 17:51:53 0 [Note] WSREP: Failed to establish connection: Connection refused
```

The pods are deleted, and then one or more (usually just one) recovery pods start up and successfully finish:
```
2024-11-08 17:47:39 0 [Note] slave_connections_needed_for_purge changed to 0 because of Galera. Change it to 1 or higher if this Galera node is also Master in a normal replication setup
2024-11-08 17:47:39 0 [Note] Starting MariaDB 11.4.3-MariaDB-ubu2404-log source revision 5ab81ffe0097a22a774957df28c5223cf0201de3 server_uid ToG1RgdGosrxu85y06fMwM44foQ= as process 1
2024-11-08 17:47:39 0 [Note] InnoDB: Compressed tables use zlib 1.3
2024-11-08 17:47:39 0 [Note] InnoDB: Number of transaction pools: 1
2024-11-08 17:47:39 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-11-08 17:47:39 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
2024-11-08 17:47:39 0 [Warning] mariadbd: io_uring_queue_init() failed with errno 95
2024-11-08 17:47:39 0 [Warning] InnoDB: liburing disabled: falling back to innodb_use_native_aio=OFF
2024-11-08 17:47:39 0 [Note] InnoDB: Initializing buffer pool, total size = 1.000GiB, chunk size = 16.000MiB
2024-11-08 17:47:39 0 [Note] InnoDB: Completed initialization of buffer pool
2024-11-08 17:47:39 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
2024-11-08 17:47:39 0 [Note] InnoDB: End of log at LSN=1589997438
2024-11-08 17:47:39 0 [Note] InnoDB: Opened 3 undo tablespaces
2024-11-08 17:47:39 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
2024-11-08 17:47:39 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-11-08 17:47:39 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-11-08 17:47:39 0 [Note] InnoDB: log sequence number 1589997438; transaction id 3441
2024-11-08 17:47:39 0 [Warning] InnoDB: Skipping buffer pool dump/restore during wsrep recovery.
2024-11-08 17:47:39 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-11-08 17:47:42 0 [Note] Server socket created on IP: '0.0.0.0'.
2024-11-08 17:47:42 0 [Note] Server socket created on IP: '::'.
2024-11-08 17:47:42 0 [Note] WSREP: Recovered position: 7ae1d2a7-9df3-11ef-8950-0b390d5970b0:2039,0-1-2033
```

Then nothing happens from here. I see a `PodDisruptionObject` that is in error state because of `insufficientPods`. I deleted the PDB, but nothing else seems to happen. 

Any ideas?

**Comment by kenlasko** [comment]
I've gotten somewhere with this. Despite what the docs say, it appears that I have to set `safe_to_bootstrap: 1` on a node for the recovery to work properly. Once I do that, the cluster recovers nicely. 

Is this expected behaviour?

**Comment by mmontes11** [comment]
> I've gotten somewhere with this. Despite what the docs say, it appears that I have to set safe_to_bootstrap: 1 on a node for the recovery to work properly. Once I do that, the cluster recovers nicely. Is this expected behaviour?

No, that's precisely what we try to avoid, having to manually tweak the Galera config file.

> "Error reconciling Galera: error recovering cluster: error performing recovery: 1 error occurred:\n\t* error performing recovery in Pod 'mariadb-0': context deadline exceeded\n\n"

I think your recovery Jobs might be stucked for a different reason, which aligns with your `MariaDB` status, as it lacks sequence numbers:

>   galeraRecovery:
    state:
      mariadb-0:
        safeToBootstrap: false
        seqno: -1
        uuid: 7ae1d2a7-9df3-11ef-8950-0b390d5970b0
        version: '2.1'
      mariadb-1:
        safeToBootstrap: false
        seqno: -1
        uuid: 7ae1d2a7-9df3-11ef-8950-0b390d5970b0
        version: '2.1'
      mariadb-2:
        safeToBootstrap: false
        seqno: -1
        uuid: 7ae1d2a7-9df3-11ef-8950-0b390d5970b0
        version: '2.1'

I would inspect the recovery Job logs to see if there is something outstanding.
